### PR TITLE
Additional FIO messages (renew_fio_address, new_funds_request)

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Extend FIO support with additional messages:
- renewal of FIO address (renew_fio_address)
- new payment request (new_funds_request)
Solves #850 .

## Testing instructions

Unit tests.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

File: src/proto/FIO.proto [hunk 1]
```diff
@@ -62,10 +62,49 @@ message Action {
...
+        string fio_address = 1;
+
+        // FIO address of the owner. Ex.: "FIO6m1fMdTpRkRBnedvYshXCxLFiC5suRU8KDfx8xxtXp2hntxpnf"
+        string owner_fio_public_key = 2;
+        
+        // Max fee to spend, can be obtained using get_fee API.
+        uint64 fee = 3;
+
+        // The FIO name of the originating wallet (project-wide constant)
+        string tpid = 4;
```
Comment (hewigovens) [inline src/proto/FIO.proto]: let's move tpid to `SigningInput` level?

File: src/FIO/TransactionBuilder.h [hunk 2]
```diff
@@ -31,35 +31,35 @@ class TransactionBuilder {
 public:
     static constexpr auto ApiAccountAddress = "fio.address";
     static constexpr auto ApiAccountToken = "fio.token";
+    static constexpr auto ApiAccountPayRequest = "fio.reqobt";
```
Comment (hewigovens) [inline src/FIO/TransactionBuilder.h]: I noticed this actually is an eos smart contract, same for `fio.token`, `fio.address`, maybe `ContractPayRequest` is more meaningful

File: src/proto/FIO.proto (hunk 1 above)
Comment (catenocrypt) [reply src/proto/FIO.proto]: Good idea, done.

File: src/FIO/TransactionBuilder.h (hunk 2 above)
Comment (catenocrypt) [reply src/FIO/TransactionBuilder.h]: done

**Comment by hewigovens** [comment]
could you please handle `TW_FIO_Proto_SigningInput` in `FIO::Signer`? (not `TWFIOSigner`)? for new `AnySigner`

**Comment by catenocrypt** [comment]
> could you please handle `TW_FIO_Proto_SigningInput` in `FIO::Signer`? (not `TWFIOSigner`)? for new `AnySigner`

Makes perfect sense, I had some subconscious anxiety about having logic in the TW...Signer.  Changing it.

**Pull request merged**